internal/provider/vdc/vdc_group_datasource.go

```diff
@@ -0,0 +1,275 @@
+						"is_remote_org": schema.BoolAttribute{
+							MarkdownDescription: "Specifies whether the vDC is local to this VCD site",
+							Computed:            true,
+						},
+						"status": schema.StringAttribute{
+							MarkdownDescription: "The status that the vDC can be in 'SAVING', 'SAVED', 'CONFIGURING'," +
+								" 'REALIZED', 'REALIZATION_FAILED', 'DELETING', 'DELETE_FAILED', 'OBJECT_NOT_FOUND'," +
+								" 'UNCONFIGURED'",
+							Computed: true,
+						},
+						"site_name": schema.StringAttribute{
+							MarkdownDescription: "Site vDC belongs.",
+							Computed:            true,
+						},
+						"site_id": schema.StringAttribute{
+							MarkdownDescription: "Site vDC belongs.",
+							Computed:            true,
+						},
+						"name": schema.StringAttribute{
+							MarkdownDescription: "vDC name.",
+							Computed:            true,
+						},
+						"id": schema.StringAttribute{
+							MarkdownDescription: "vDC ID",
+							Computed:            true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *vdcGroupDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.CloudAvenue)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.CloudAvenue, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *vdcGroupDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data vdcGroupDataSourceModel
+	var diag diag.Diagnostics
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	adminOrg, err := d.client.Vmware.GetAdminOrgByNameOrId(d.client.GetOrg())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving Org", err.Error())
+		return
+	}
+
+	// Get the vdcGroup. If id is not set, user name.
+	var vdcGroup *govcd.VdcGroup
+	if !data.ID.IsNull() && !data.ID.IsUnknown() {
+		vdcGroup, err = adminOrg.GetVdcGroupById(data.ID.ValueString())
+	} else {
+		vdcGroup, err = adminOrg.GetVdcGroupByName(data.Name.ValueString())
+	}
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Error retrieving vDC Group", err.Error())
+		return
+	}
+
+	// populate the data source model with the vdcGroup data
+	data = vdcGroupDataSourceModel{
+		ID:                         types.StringValue(vdcGroup.VdcGroup.Id),
+		Name:                       types.StringValue(vdcGroup.VdcGroup.Name),
+		Description:                types.StringValue(vdcGroup.VdcGroup.Description),
+		DfwEnabled:                 types.BoolValue(vdcGroup.VdcGroup.DfwEnabled),
```

Comment:
`DFW` is acronym. I suggest to UpperCase `DFW` and add this rule in `.golangci.yml`